C error: string/char type mismatch on map, but still compiles

```
v -debug main.v
.main.c: In function ‘increment_string’:
.main.c:3323:2: error: aggregate value used where an integer was expected
  if ( lastcharacter == ((byte)( tos2((byte*)"9") ) ) ) {
  ^~
.main.c:3343:64: error: incompatible type for argument 2 of ‘map_get’
  string tmp8 = tos("", 0); bool tmp9 = map_get( incrementmap , lastcharacter, & tmp8);
                                                                ^~~~~~~~~~~~~
.main.c:3164:29: note: expected ‘string’ {aka ‘struct string’} but argument is of type ‘byte’ {aka ‘unsigned char’}
  bool map_get(map m, string key, void* out) {
                      ~~~~~~~^~~
V panic: C error. This should never happen. Please create a GitHub issue: https://github.com/vlang/v/issues/new/choose
```

[main(c).txt](https://github.com/vlang/v/files/3462308/main.c.txt)
[main(v).txt](https://github.com/vlang/v/files/3462309/main.v.txt)

--- a/compiler/parser.v
+++ b/compiler/parser.v
@@ -1969,7 +1969,11 @@ fn (p mut Parser) index_expr(typ_ string, fn_ph int) string {
 			}
 		}
 		else {
-			p.expression()
+			T := p.table.find_type(p.expression())
+			// TODO: Get the key type of the map instead of only string.
+			if is_map && T.parent != 'string' {
+				p.check_types(T.name, 'string')
+			}
 		}
 		p.check(.rsbr)
 		// if (is_str && p.builtin_mod) || is_ptr || is_fixed_arr && ! (is_ptr && is_arr) {

